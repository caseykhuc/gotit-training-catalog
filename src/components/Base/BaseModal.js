import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'
import { hideModal } from '../../actions/modal';

export const BaseModal = ({
  title, hideModal, onAccept, children,
}) => (
    <Modal show onHide={hideModal} centered>
      <Modal.Header closeButton>
        <Modal.Title>{title}</Modal.Title>
      </Modal.Header>
      <Modal.Body>{children}</Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" onClick={hideModal}>
          Close
      </Button>
        <Button variant="primary" onClick={onAccept}>
          Accept
      </Button>
      </Modal.Footer>
    </Modal>
  )

BaseModal.propTypes = {
  title: PropTypes.string.isRequired,
  hideModal: PropTypes.func.isRequired,
  onAccept: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
}

export default connect(null, { hideModal })(BaseModal);
