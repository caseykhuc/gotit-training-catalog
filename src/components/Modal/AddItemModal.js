import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as selector from 'reducers';
import { addItem } from 'actions/item';
import BaseFormModal from 'components/Base/BaseFormModal';

import validator from 'validator';
import * as errorMessage from 'utils/inputError';

const validate = ({
  name, price, categoryId,
}) => {
  const inputError = {};
  if (name && name.length < 5) inputError.name = errorMessage.name.tooShort;
  if (price && !validator.isNumeric(price)) {
    inputError.price = errorMessage.price.tooSimple;
  }
  if (!categoryId) {
    inputError.categoryId = errorMessage.category.isNotDefined;
  }
  return inputError;
};

export const AddItemModal = ({ addItem, categories }) => {
  const initialState = {
    inputValue: {
      name: '',
      description: '',
      price: '',
      categoryId: categories[0],
    },
    inputError: {
    },
    requestError: '',
  };

  const fields = [
    { name: 'name', type: 'text' },
    { name: 'description', type: 'text' },
    { name: 'price', type: 'text' },
    { name: 'categoryId', type: 'select', options: categories },
  ];

  return (
    <BaseFormModal
      title="ADD ITEM"
      fields={fields}
      initialState={initialState}
      onAction={addItem}
      validate={validate}
    />
  )
}

AddItemModal.propTypes = {
  addItem: PropTypes.func.isRequired,
  categories: PropTypes.arrayOf(PropTypes.object).isRequired,
}

const mapStateToProps = (state) => ({
  categories: selector.getCategories(state),
})

export default connect(mapStateToProps, { addItem })(AddItemModal);
