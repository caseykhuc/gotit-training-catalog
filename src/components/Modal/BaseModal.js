import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import { hideModal } from '../../actions';
import { connect } from 'react-redux';

class BaseModal extends React.Component {
  render() {
    return (
      <Modal.Dialog>
        <Modal.Header closeButton onClick={this.props.hideModal}>
          <Modal.Title>Modal title</Modal.Title>
        </Modal.Header>

        <Modal.Body>
          <p>Modal body text goes here.</p>
        </Modal.Body>

        <Modal.Footer>
          <Button variant='secondary'>Close</Button>
          <Button variant='primary'>Save changes</Button>
        </Modal.Footer>
      </Modal.Dialog>
    );
  }
}

export default connect(null, { hideModal })(BaseModal);
