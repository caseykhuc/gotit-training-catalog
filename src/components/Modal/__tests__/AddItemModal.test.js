import React from 'react';
import { shallow } from 'enzyme';
import { AddItemModal } from '../AddItemModal';
import BaseModal from '../../Base/BaseModal';

describe('components/Modal/AddItemModal', () => {
  let props;
  let wrapper;
  let baseModal;

  const update = () => {
    wrapper.update();
    baseModal = wrapper.find(BaseModal);
  };
  const setup = () => {
    wrapper = shallow(<AddItemModal {...props} />);
    update();
  };
  beforeEach(() => {
    props = {
      addItem: jest.fn().mockResolvedValue(),
      categories: [{}],
    }
  });
  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
})
