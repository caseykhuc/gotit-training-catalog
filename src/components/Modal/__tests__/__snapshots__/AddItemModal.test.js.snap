// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Modal/AddItemModal should render correctly 1`] = `ShallowWrapper {}`;

exports[`components/Modal/DeleteItemModal should render correctly 1`] = `ShallowWrapper {}`;
